cmd/xgql/main.go

```diff
@@ -104,19 +108,44 @@ func main() {
 	log := logging.NewLogrLogger(zl.WithName("xgql"))
 
 	kingpin.FatalIfError(otelruntime.Start(), "cannot add OpenTelemetry runtime instrumentation")
+
+	// Observing a log line during startup as follows:
+	// 2021/06/08 10:03:26 <nil>
+	// This appears to be fixed with https://github.com/open-telemetry/opentelemetry-go/pull/1851
+	// However, there is no release for opentelemetry-go with this fix yet.
 	res := resource.NewWithAttributes(attribute.String("service.name", "crossplane.io/gql"))
 
 	// OpenTelemetry metrics.
 	prom, err := prometheus.InstallNewPipeline(prometheus.Config{}, basic.WithResource(res))
 	kingpin.FatalIfError(err, "cannot create OpenTelemetry Prometheus exporter")
 
-	// OpenTelemetry tracing.
-	if *agent != "" {
-		flush, err := jaeger.InstallNewPipeline(jaeger.WithAgentEndpoint(*agent), jaeger.WithSDKOptions(
-			trace.WithSampler(trace.ParentBased(trace.TraceIDRatioBased(*ratio))),
-			trace.WithResource(res)))
+	tpOpts := []trace.TracerProviderOption{
+		trace.WithSampler(trace.ParentBased(trace.TraceIDRatioBased(*ratio))),
+		trace.WithResource(res),
+	}
+	switch *tracer {
+	case "jaeger":
+		log.Debug("Enabling Jaeger tracer")
+		if *jaegerAgent == "" {
+			kingpin.Fatalf("cannot enable Jaeger tracing, jaeger agent not set")
+		}
+		host, port, err := net.SplitHostPort(*jaegerAgent)
+		kingpin.FatalIfError(err, "cannot parse jaeger agent as host:port")
```

Comment:
Hrm it's kind of annoying that the jaeger API changed completely with 0.20. I believe we could use a `TCPVar()` flag instead of a `String()` flag in order to validate that value was a `host:port`. The only minor downside there is that that flag resolves the host to an IP at flag parsing time.